ArduCopter/mode_auto.cpp

```diff
@@ -643,66 +650,65 @@ void Copter::set_auto_yaw_mode(uint8_t yaw_mode)
 
     case AUTO_YAW_RATE:
         // initialise target yaw rate to zero
-        auto_yaw_rate_cds = 0.0f;
+        _rate_cds = 0.0f;
         break;
     }
 }
 
-// set_auto_yaw_look_at_heading - sets the yaw look at heading for auto mode
-void Copter::set_auto_yaw_look_at_heading(float angle_deg, float turn_rate_dps, int8_t direction, bool relative_angle)
+// set_fixed_yaw - sets the yaw look at heading for auto mode
+void Copter::Mode::_AutoYaw::set_fixed_yaw(float angle_deg, float turn_rate_dps, int8_t direction, bool relative_angle)
 {
-    // get current yaw target
-    int32_t curr_yaw_target = attitude_control->get_att_target_euler_cd().z;
+    const int32_t curr_yaw_target = copter.attitude_control->get_att_target_euler_cd().z;
 
     // calculate final angle as relative to vehicle heading or absolute
     if (!relative_angle) {
         // absolute angle
-        yaw_look_at_heading = wrap_360_cd(angle_deg * 100);
+        _fixed_yaw = wrap_360_cd(angle_deg * 100);
     } else {
         // relative angle
         if (direction < 0) {
             angle_deg = -angle_deg;
         }
-        yaw_look_at_heading = wrap_360_cd((angle_deg * 100) + curr_yaw_target);
+        _fixed_yaw = wrap_360_cd((angle_deg * 100) + curr_yaw_target);
     }
 
     // get turn speed
     if (is_zero(turn_rate_dps)) {
         // default to regular auto slew rate
-        yaw_look_at_heading_slew = AUTO_YAW_SLEW_RATE;
+        fixed_yaw_slewrate = AUTO_YAW_SLEW_RATE;
     } else {
-        int32_t turn_rate = (wrap_180_cd(yaw_look_at_heading - curr_yaw_target) / 100) / turn_rate_dps;
-        yaw_look_at_heading_slew = constrain_int32(turn_rate, 1, 360);    // deg / sec
+        const int32_t turn_rate = (wrap_180_cd(_fixed_yaw - curr_yaw_target) / 100) / turn_rate_dps;
+        _fixed_yaw_slewrate = constrain_int32(turn_rate, 1, 360);    // deg / sec
     }
 
     // set yaw mode
-    set_auto_yaw_mode(AUTO_YAW_LOOK_AT_HEADING);
+    set_mode(AUTO_YAW_FIXED);
 
     // TO-DO: restore support for clockwise and counter clockwise rotation held in cmd.content.yaw.direction.  1 = clockwise, -1 = counterclockwise
 }
 
-// set_auto_yaw_roi - sets the yaw to look at roi for auto mode
-void Copter::set_auto_yaw_roi(const Location &roi_location)
+// set_roi - sets the yaw to look at roi for auto mode
+void Copter::Mode::_AutoYaw::set_roi(const Location &roi_location)
 {
     // if location is zero lat, lon and altitude turn off ROI
     if (roi_location.alt == 0 && roi_location.lat == 0 && roi_location.lng == 0) {
         // set auto yaw mode back to default assuming the active command is a waypoint command.  A more sophisticated method is required to ensure we return to the proper yaw control for the active command
-        set_auto_yaw_mode(get_default_auto_yaw_mode(false));
+        auto_yaw.set_mode_to_default(false);
```

Comment:
The auto_yaw isn't needed right? This method belongs to the class now. There are others like this below.